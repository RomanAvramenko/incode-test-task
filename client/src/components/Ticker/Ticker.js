import React from "react";
import "./Ticker.scss";

export const Ticker = ({ data, comparison }) => {
  const arrowDown = (
    <path d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"></path>
  );

  const arrowUp = (
    <path d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"></path>
  );

  let arrowIcon = (
    <path d="M16.79 7.79l-1.41 1.42L17.17 11H3v2h14.17l-1.79 1.79 1.41 1.42L21 12z"></path>
  );

  let color = "color_default";
  let sign;

  if (comparison !== undefined) {
    color = comparison ? "color_fall" : "color_grow";
    sign = comparison ? "-" : "+";
    arrowIcon = comparison ? arrowDown : arrowUp;
  }

  return (
    <div className={`ticker_item ${color}`}>
      <span className="ticker_item_heading">{data.exchange}</span>
      <div className="ticker_item_wrapper">
        <span>
          <div>
            <div className={`ticker_item_icon ${color}`}>
              <span className="ticker_item_icon-container" aria-hidden="true">
                <svg
                  focusable="false"
                  width="16"
                  height="16"
                  viewBox="0 0 24 24"
                  className="ticker_item_icon-container arrow"
                >
                  {arrowIcon}
                </svg>
              </span>
            </div>
          </div>
        </span>
        <div className="ticker_item_main">
          <div className="ticker_item_main_name">{data.ticker}</div>
          <div className="ticker_item_main_price">{data.price}</div>
        </div>
        <div className="ticker_item_changes">
          <span>
            {sign}
            {data.change_percent}%
          </span>
          <span>
            {sign}
            {data.change}
          </span>
        </div>
      </div>
    </div>
  );
};
