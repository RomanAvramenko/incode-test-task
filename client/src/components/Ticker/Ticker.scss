.color {
  &_grow {
    color: #137333;
    background: #e6f4ea;
  }
  &_fall {
    color: #c5221f;
    background: #fce8e6;
  }
  &_default {
    background: #f8f9fa;
    color: #3c4043;
  }
}

.ticker_item {
  display: flex;
  flex-direction: column;
  padding: 6px;
  background-color: white;
  border-radius: 4px;
  border: 1px solid #dadce0;
  white-space: nowrap;

  &_heading {
    color: #202124;
    font-size: 0.8em;
    margin-bottom: 1em;
    text-align: center;
  }

  &_icon {
    box-align: center;
    align-items: center;
    height: 32px;
    width: 32px;
    min-width: 32px;
    border-radius: 8px;
    margin-right: 8px;
    display: flex;
    &-container {
      height: 16px;
      margin: auto;
      .arrow {
        fill: currentColor;
        flex-shrink: 0;
      }
    }
  }

  &_wrapper {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }

  &_main {
    display: flex;
    flex-direction: column;
    letter-spacing: 0.025em;
    font-family: Roboto, Arial, sans-serif;
    font-size: 0.75rem;
    font-weight: 400;
    line-height: 1rem;
    margin-left: -2em;
    max-width: 120px;
    &_name {
      color: #202124;
      font-weight: bold;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }
    &_price {
      color: #3c4043;
    }
  }
  &_changes {
    display: flex;
    -webkit-box-orient: vertical;
    box-orient: vertical;
    flex-direction: column;
    letter-spacing: 0.025em;
    font-family: Roboto, Arial, sans-serif;
    font-size: 0.75rem;
    font-weight: 400;
    line-height: 1rem;
    text-align: right;
  }
}
