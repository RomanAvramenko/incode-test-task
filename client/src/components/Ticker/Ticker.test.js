import React from "react";
import { shallow } from "enzyme";
import { Ticker } from "./Ticker";

const mockProps = {
  data: {
    ticker: "Ticker_Mock",
    price: 1,
    change_percent: 1.1,
    change: 0.1,
  },
};

describe("Ticker component", () => {

  it("should render Ticker component", () => {
    const component = shallow(<Ticker {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it('Ticker element exist in the DOM',()=>{
    const component = shallow(<Ticker {...mockProps} />);
    const wrapper = component.find(".ticker_item");
    expect(wrapper.length).toBe(1);
  });
});
