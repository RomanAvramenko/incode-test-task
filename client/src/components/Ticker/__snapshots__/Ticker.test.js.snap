// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticker component should render Ticker component 1`] = `
<div
  className="ticker_item color_default"
>
  <span>
    <div>
      <div
        className="ticker_item_icon color_default"
      >
        <span
          aria-hidden="true"
          className="ticker_item_icon-container"
        >
          <svg
            className="ticker_item_icon-container arrow"
            focusable="false"
            height="16"
            viewBox="0 0 24 24"
            width="16"
          >
            <path
              d="M16.79 7.79l-1.41 1.42L17.17 11H3v2h14.17l-1.79 1.79 1.41 1.42L21 12z"
            />
          </svg>
        </span>
      </div>
    </div>
  </span>
  <div
    className="ticker_item_wrapper"
  >
    <div
      className="ticker_item_main"
    >
      <div
        className="ticker_item_main_name"
      >
        Ticker_Mock
      </div>
      <div
        className="ticker_item_main_price"
      >
        1
      </div>
    </div>
    <div
      className="ticker_item_changes"
    >
      <span>
        1.1
        %
      </span>
      <span>
        0.1
      </span>
    </div>
  </div>
</div>
`;
